frontend/src/components/Board/BoardComponents/TaskBoard.tsx

```diff
@@ -1,55 +1,170 @@
-  ));
+export function TaskBoard({ board, onEditCardSelect }: IBoardProp) {
+  const supabase = useClient();
+  const [columns, setColumns] = useState<Column[] | undefined | null>();
+  const [cards, setCards] = useState<Card[]>([]);
+  // Get Columns
+  useEffect(() => {
+    getColumns();
+  }, [board]);
+
+  async function getColumns() {
+    const { data } = await supabase
+      .from('list')
+      .select()
+      .match({ board_id: board.id, board_version: board.version })
+      .order('index', { ascending: true });
+    setColumns(data);
+  }
+  // Get Cards of each Column
+  useEffect(() => {
+    getColumnCards();
+  }, [columns]);
+
+  function getColumnCards() {
+    if (!columns) {
+      return;
+    }
+    // setCards([]);
+    let count = 0;
+    const card_list: Card[] = [];
+    columns?.forEach((col) => {
+      // console.log(col.id)
+      getCards(col).then((result) => {
+        // console.log('result', result)
+        result?.forEach((card) => {
+          card_list.push(card);
+        });
+        count++;
+        if (count == columns?.length) {
+          setCards(card_list);
+        }
+      });
+    });
+  }
+
+  async function getCards(col: Column) {
+    const { data } = await supabase
+      .from('card')
+      .select()
+      .match({ list_id: col.id })
+      .order('index', { ascending: true });
+    return data;
+  }
+
+  const [columnElements, setColumnElements] = useState<JSX.Element[]>();
+  useEffect(() => {
+    setColumnElements(
+      columns?.map((col) => (
+        <TaskColumn
+          column={col}
+          cards={cards
+            ?.filter((card) => {
+              return card.list_id == col.id;
+            })
+            .sort((card1, card2) => {
+              return card1.index - card2.index;
+            })}
+          onEditCardClick={onEditCardSelect}
+        />
+      )),
+    );
+    // console.log({ cards, columns})
+  }, [columns, cards]);
+
+  if (!columns) {
+    return (
+      <>
+        <div className="bg-gray-500 h-full flex flex-col justify-center items-center">
+          No Columns Found
+        </div>
+      </>
+    );
+  }
 
   // Handles moving cards
   function onDragEnd(result: DropResult) {
     // console.log('drag ended ', result);
-    const columnSource: IColumn | undefined = columns?.find(
-      (c) => c.id == result?.source?.droppableId,
-    );
-    const columnDestination: IColumn | undefined = columns?.find(
-      (c) => c.id == result?.destination?.droppableId,
-    );
-    const cardToMove: ICard | undefined = columnSource?.cards?.find(
-      (c) => c.id == result?.draggableId,
-    );
-    if (cardToMove && columnSource && columnDestination) {
-      cardToMove.list_id = result?.destination?.droppableId;
-      // If moving cards in the same column
-      if (result?.destination?.index == null) {
-        return;
+    if (!result?.destination?.droppableId || !result.draggableId) {
```

Comment:
Can we pull out all the conditions into variables, it will make it easier to read in the future. And it would be the practice of self-documenting code.
Example:
The var name might be wrong if I don't understand the condition correctly, but you should get the idea of what I'm talking about. You can do this for the rest of conditions in this function. Would make it much more readable.
`const isDragNotFinished = !result?.destination?.droppableId || !result.draggableId`